src/components/Navbar/LanguageSelector.tsx

```diff
@@ -0,0 +1,48 @@
+import { MenuItem, TextField } from '@mui/material';
+import { useContext } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import GeneralContext from '../../context/GeneralContext';
+import { GeneralContextAction } from '../../types/GeneralContextType';
+
+function LanguageSelector() {
+  const { i18n } = useTranslation();
+  const generalContext = useContext(GeneralContext);
+
+  if (!generalContext) {
+    return;
+  }
+
+  function handleLanguageChange(lang: string) {
+    generalContext!.dispatch({
```

Comment:
use **if check** here instate of directly doing **generalContext!** .
```suggestion
if(generalContext) {
    generalContext.dispatch({
    }
```

Reply:
Done that.